adapter.aasx: Add `rename_file` to `DictSupplementaryFileContainer`

thammel opened the pull request:
Before:
There was no supported way to rename supplementary files in `DictSupplementaryFileContainer`. Workflow had to manipulate internals or create duplicates, making file handling error-prone.

Now:
A public rename operation preserves content de-duplication and resolves name conflicts, simplifying AASX read/write flows and reducing inconsistent mappings.

See [rwth.aachen/aas-manager#46](https://github.com/rwth-iat/aas_manager/issues/46)

moritzsommer committed:
Small refactoring

thammel committed:
adapter.aasx: Add `rename_file` to `DictSupplementaryFileContainer`

Before:
There was no supported way to rename supplementary files in
`DictSupplementaryFileContainer`. Workflow had to manipulate internals or create duplicates, making file handling error-prone.

Now:
A public rename operation preserves content de-duplication and resolves name conflicts, simplifying AASX read/write flows and reducing inconsistent mappings.

s-heppner commented:
Thank you, that's a good idea! Could you please also add a unittest for your new method to [this](https://github.com/eclipse-basyx/basyx-python-sdk/blob/4bd142a155d433d02f9e6b31ffb2185d9b56ca0c/sdk/test/adapter/aasx/test_aasx.py#L21) `TestCase` class?

And while you're add it, if you merge `develop` into your branch (should not have any conflicts, I think), then the `check-python-versions` CI should pass again.

thammel committed:
erge remote-tracking branch 'upstream/develop' into add/rename-file
Merge develop branch

This should resolve CI errors
Merge remote-tracking branch 'upstream/develop' into add/rename-file

zrgt commented on sdk/basyx/aas/adapter/aasx.py:
```diff
@@ -824,15 +824,24 @@ def add_file(self, name: str, file: IO[bytes], content_type: str) -> str:
+            raise KeyError(f"File with name {old_name} not found in SupplementaryFileContainer.")
+        if new_name == old_name:
+            return new_name
+        old_hash, old_ct = self._name_map[old_name]
```
Rename to file_hash and file_ct, as these are not old, these are still actual

zrgt commented on sdk/basyx/aas/adapter/aasx.py:
```diff
@@ -824,15 +824,24 @@ def add_file(self, name: str, file: IO[bytes], content_type: str) -> str:
             if new_name not in self._name_map:
-                self._name_map[new_name] = name_map_data
-                self._store_refcount[hash] += 1
+                self._name_map[new_name] = (sha, content_type)
```
Here as proposed, we should delete the old dict item with `old_name`

thammel committed:
test_aasx: Add unit tests for rename_file method

thammel committed:
Fix rename_file method to delete old name from name map

zrgt committed:
Small refactoring

zrgt committed:
Small refactoring
